server/src/service/BoardService.js

```diff
@@ -10,4 +11,32 @@ export class BoardService extends BaseService {
 
         return BoardService.instance;
     }
+
+    async getMyBoards(id) {
+        const boards = await this.boardRepository.find({
+            select: ['id', 'title'],
+            where: { creator: id },
+        });
+        return boards;
+    }
+
+    async getInvitedBoards(id) {
+        const boards = await this.boardRepository
+            .createQueryBuilder('board')
+            .select('board.id')
+            .addSelect('board.title')
+            .innerJoin('board.invitations', 'invitation', 'invitation.user=:userId', { userId: id })
```

Comment:
```suggestion
            .innerJoin('board.invitations', 'a', 'a.user_id=:userId', { userId: id })
```
테스트를 돌려보지 않아서 확실치 않지만, 위처럼 해야지 동작할 것 같아요.
Join 성능을 생각했을 때, board 보다는 invitation을 먼저 조회하는 것이 더 빠를 것 같다는 생각이 듭니다.
board를 먼저 조회하게 되면 invitation이 없는 board도 불필요하게 검색할 것 같아요.

Reply:
model 폴더에서 정의한 엔티티의 속성대로 컬럼명이 따라가는 것으로 이해를 했었습니다.🤔
invitation 엔티티에서 user_id 컬럼이 user로 정의되어 있어 user로 사용했는데 이거는 쿼리문을 한번 확인해봐야겠네요.

Reply:
일단 제가 실행해보았을 때, 제가 작성한 것과 건홍님이 작성한 것 모두 아래와 같은 동일한 쿼리가 만들어지더라구요.
```sql
SELECT "board"."id" AS "board_id", "board"."title" AS "board_title" FROM "board" "board"
INNER JOIN "invitation" "invitation" ON "invitation"."board_id"="board"."id" AND ("invitation"."user_id"=?)
```
신기하네요. typeORM👀
쿼리는 동일하지만 코드상으로 건홍님 코드가 좀 더 직관적일 것 같다는 느낌이 들어서 join 순서 변경하면서 수정하도록 하겠습니다.

Reply:
join 성능에 대한 말씀을 하셔서 inner join 성능에 대해서 이래저래 구글링 해보았습니다.
아래 글에 의하면 MySQL의 경우, 두개의 테이블 inner join은 테이블 순서에 상관없이 성능이 동일할 것이라는 생각이 드네요.
[MySQL 테이블 조인](https://12bme.tistory.com/165)

다만 세개 이상의 테이블을 inner join하는 경우에는 join 순서가 성능에 영향을 미치는 것 같네요.
[쿼리 최적화](https://medium.com/watcha/%EC%BF%BC%EB%A6%AC-%EC%B5%9C%EC%A0%81%ED%99%94-%EC%B2%AB%EA%B1%B8%EC%9D%8C-%EB%B3%B4%EB%8B%A4-%EB%B9%A0%EB%A5%B8-%EC%BF%BC%EB%A6%AC%EB%A5%BC-%EC%9C%84%ED%95%9C-7%EA%B0%80%EC%A7%80-%EC%B2%B4%ED%81%AC-%EB%A6%AC%EC%8A%A4%ED%8A%B8-bafec9d2c073) -> 6번 문항이 inner join 순서에 대한 얘기네요!

DB 성능 고려하기..헷갈리네요..😵
MySQL은 join 방식이 NL 방식으로 고정이라고 들었는데..
오라클의 경우에는 여러 join 방식을 제공한다고 해서 그렇다면 오라클에서는 join 순서에 따라 더 많은 성능 차이가 나려나요.. 궁금해지네요🤔

Reply:
좋은 글 감사합니다. 도움이 많이 됐어요👍
`("invitation"."user_id"=?)` 와 같은 추가적인 작업이 있어서 많이 헷갈렸는데 시원하게 해결됐습니다.
나중에 board_id와 user_id를 다중 인덱스 걸면 더 빠르게 찾을 수 있을 듯 합니다!